x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/pki/PkiRealm.java

```diff
@@ -231,6 +236,28 @@ public void lookupUser(String username, ActionListener<User> listener) {
         listener.onResponse(null);
     }
 
+    String getPrincipalFromToken(X509AuthenticationToken token) {
+        return principalRdnEnabled
+            ? getPrincipalFromRdnAttribute(principalRdnType, token, logger)
+            : getPrincipalFromSubjectDN(principalPattern, token, logger);
+    }
+
+    static String getPrincipalFromRdnAttribute(String principalRdnType, X509AuthenticationToken token, Logger logger) {
+        X500Principal certPrincipal = token.credentials()[0].getSubjectX500Principal();
+        String principal = RdnFieldExtractor.extract(certPrincipal.getEncoded(), principalRdnType);
+        if (principal == null) {
+            logger.debug(
```

Comment:
Maybe this should be a warning since it's probably due to a misconfiguration most of the time?